internal/rest/k8s_utils.go

```diff
@@ -110,10 +110,41 @@ func (rest *RestOperations) SyncObjectStatuses() {
 	}
 
 	if lib.GetAdvancedL4() {
-		status.UpdateGatewayStatusAddress(allGatewayUpdateOptions, true)
+		for i := range allGatewayUpdateOptions {
+			statusOption := status.StatusOptions{
+				ObjType: lib.Gateway,
+				Op:      lib.UpdateOperation,
+				Options: &allGatewayUpdateOptions[i],
+			}
+			statusQueue := utils.SharedWorkQueue().GetQueueByName(utils.StatusQueue)
+			bkt := utils.Bkt(allGatewayUpdateOptions[i].ServiceMetadata.Gateway, statusQueue.NumWorkers)
+			statusQueue.Workqueue[bkt].AddRateLimited(statusOption)
```

Comment:
i guess these can be moved to a separate PublishToStatusQueue function?

Reply:
Done